testsuite/gateways/apicast/operator.py

```diff
@@ -218,3 +218,55 @@ def _add_jaeger(apicast):
         self.apicast.modify_and_apply(_add_jaeger)
         self.reload()
         return self.name
+
+    def __getstate__(self):
+        """
+        Custom serializer for pickle module
+        more info here: https://docs.python.org/3/library/pickle.html#object.__getstate__
+        """
+        return {
+            "CAPABILITIES": self.CAPABILITIES,
+            "HAS_PRODUCTION": self.HAS_PRODUCTION,
+            "PRIORTY": self.PRIORITY,
+            "apicast": {
+                "logLevel": self.apicast["logLevel"],
+                "openSSLPeerVerificationEnabled": self.apicast["openSSLPeerVerificationEnabled"],
+                "deploymentEnvironment": self.apicast["deploymentEnvironment"],
+                "cacheConfigurationSeconds": self.apicast["cacheConfigurationSeconds"],
+                "configurationLoadMode": self.apicast["configurationLoadMode"],
+                "image": self.apicast["image"]
+            },
+            "_deployment": self.deployment,
+            "reload": self.reload,
+            "image": self.image,
+            "name": self.name,
+            "openshift": self.openshift,
+            "path_routing": self.path_routing,
+            "portal_endpoint": self.portal_endpoint,
+            "secure": self.secure,
+            "staging": self.staging,
+            "_base_route": self._base_route,
+            "_routes": self._routes,
+            "_to_delete": self._to_delete
+        }
+
+    def __setstate__(self, state):
+        """
+        Custom deserializer for pickle module
+        more info here: https://docs.python.org/3/library/pickle.html#object.__setstate__
+        """
+        for k in state:
+            if k != "apicast" or k != "reload":
+                setattr(self, k, state[k])
+        apicast = APIcast.create_instance(
```

Comment:
What if you stored just the name of the APICast on the openshift and then used that as a base?